src/CoreVideo/CVDisplayLink.cs

```diff
@@ -192,20 +240,43 @@ public CVReturn GetCurrentTime (out CVTimeStamp outTime)
 		static CVReturn OutputCallback (IntPtr displayLink, ref CVTimeStamp inNow, ref CVTimeStamp inOutputTime, CVOptionFlags flagsIn, ref CVOptionFlags flagsOut, IntPtr displayLinkContext)
 		{
 			GCHandle callbackHandle = GCHandle.FromIntPtr (displayLinkContext);
-			DisplayLinkOutputCallback func = (DisplayLinkOutputCallback) callbackHandle.Target;
+			DisplayLinkOutputCallback func = (DisplayLinkOutputCallback) callbackHandle.Target!;
 			CVDisplayLink delegateDisplayLink = new CVDisplayLink(displayLink, false);
 			return func (delegateDisplayLink, ref inNow, ref inOutputTime, flagsIn, ref flagsOut);
 		}
 	  
 		[DllImport (Constants.CoreVideoLibrary)]
 		extern static CVReturn CVDisplayLinkSetOutputCallback (IntPtr displayLink, CVDisplayLinkOutputCallback function, IntPtr userInfo);
+
 		public CVReturn SetOutputCallback (DisplayLinkOutputCallback callback)
 		{
 			callbackHandle = GCHandle.Alloc (callback);
 			CVReturn ret = CVDisplayLinkSetOutputCallback (this.Handle, static_OutputCallback, GCHandle.ToIntPtr (callbackHandle));
 				
 			return ret;
 		}
+
+		[Mac (12,0), NoiOS, NoTV]
+		[DllImport (Constants.CoreVideoLibrary)]
+		static extern nuint CVDisplayLinkGetTypeID ();
+
+		[Mac (12,0), NoiOS, NoTV]
+		public static nuint GetTypeId ()
+			=> CVDisplayLinkGetTypeID ();
+
+		[Mac (12,0), NoiOS, NoTV]
+		[DllImport (Constants.CoreVideoLibrary)]
+		static extern int CVDisplayLinkTranslateTime (IntPtr displayLink, CVTimeStamp inTime, out CVTimeStamp outTime);
+
+		[Mac (12,0), NoiOS, NoTV]
+		public bool TryTranslateTime (CVTimeStamp inTime, out CVTimeStamp? outTime)
+		{
+			outTime = null;
+			if (CVDisplayLinkTranslateTime (this.Handle, inTime, out var translated) == 0) {
```

Comment:
The documentation says this:
```
outTime
A pointer to a CVTimeStamp structure into which the target time is written. Before calling, you must set the version field (currently 0) to indicate which version of the structure you want. You should also set the flags field to specify which representations to translate to.
```
You're passing in null, which is not what the API calls for. It looks from the documentation that this should be a `ref` parameter and not an `out` parameter, and `outTime.Version` should be 0.

Reply:
Nice catch, I'll update it, thx!